go/interpreter/lfvm/memory_test.go

```diff
@@ -185,3 +192,272 @@ func TestMemory_expandMemory_expandsMemoryOnlyWhenNeeded(t *testing.T) {
 		})
 	}
 }
+
+func TestMemory_getSlice_ErrorCases(t *testing.T) {
+	c := getEmptyContext()
+	c.gas = 0
+	m := NewMemory()
+	_, err := m.getSlice(0, 1, &c)
+	if !errors.Is(err, errOutOfGas) {
+		t.Errorf("error should be errOutOfGas, instead is: %v", err)
+	}
+	_, err = m.getSlice(math.MaxUint64-31, 32, &c)
+	if !errors.Is(err, errGasUintOverflow) {
+		t.Errorf("error should be errGasUintOverflow, instead is: %v", err)
+	}
+}
+
+func TestMemory_getSlice_properlyHandlesMemoryExpansionsAndReturnsExpectedMemory(t *testing.T) {
+
+	tests := map[string]struct {
+		offset   uint64
+		size     uint64
+		expected []byte
+	}{
+		"size zero returns empty slice": {
+			offset:   64,
+			size:     0,
+			expected: []byte{},
+		},
+		"memory does not expand when not needed": {
+			offset:   0,
+			size:     4,
+			expected: []byte{0x0, 0x01, 0x02, 0x03},
+		},
+		"memory expands when needed": {
+			offset:   4,
+			size:     5,
+			expected: []byte{0x04, 4: 0x0},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			c := getEmptyContext()
+			m := NewMemory()
+			m.store = []byte{0x0, 0x01, 0x02, 0x03, 0x04}
+			slice, err := m.getSlice(test.offset, test.size, &c)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+			if !bytes.Equal(slice, test.expected) {
+				t.Errorf("unexpected slice: %x, want: %x", slice, test.expected)
+			}
+		})
+	}
+}
+
+func TestMemory_getSlice_ExpandsMemoryIn32ByteChunks(t *testing.T) {
+	for increment := 0; increment < 10; increment++ {
+		for size := 0; size < 32; size++ {
+			for offset := 0; offset < 32; offset++ {
+				c := getEmptyContext()
+				originalGas := c.gas
+				m := NewMemory()
+				offsetIncremented := uint64(offset + 32*increment)
+				_, err := m.getSlice(offsetIncremented, uint64(size), &c)
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				want := uint64(32 * (increment + 1))
+				if size == 0 {
+					want = 0
+					if c.gas != originalGas {
+						t.Error("no gas should have been consumed when size is zero.")
+					}
+				}
+				if size+offset > 32 {
+					want = uint64(32 * (increment + 2))
+				}
+				if m.length() != want {
+					t.Errorf("unexpected slice length: %d, want: %d", m.length(), 32*increment)
+				}
+			}
+		}
+	}
+}
```

Comment:
I reviewed this tests also in #761 -- please get your PRs in sync and only request reviews for relevant ones.